### PR TITLE
fix isSameOrNewer version comparison

### PR DESCRIPTION
**Pull request by boazavital**
Correctly compare semantic versions

### CODE REVIEW

**Comment by boazavital** [comment]
Added a unit test class for VersionHelper. 

Before the change, it fails with 
```
java.lang.AssertionError: expected 4.1.1.isSameOrNewer(2.6.0) to be true 
Expected :true
Actual   :false
```

File: affinity/src/main/java/net/openhft/affinity/impl/VersionHelper.java [hunk 1]
```diff
@@ -70,9 +70,12 @@ public boolean majorMinorEquals(final VersionHelper ver) {
...
         return ver != null
-                && this.major >= ver.major
-                && this.minor >= ver.minor
-                && this.release >= ver.release;
+                && (this.major > ver.major
+                    || (this.major == ver.major
+                        && this.minor > ver.minor)
+                    || (this.major == ver.major
+                        && this.minor == ver.minor
+                        && this.release >= ver.release));
```
Comment (dpisklov) [inline affinity/src/main/java/net/openhft/affinity/impl/VersionHelper.java]: could do even simpler, in same style:
```java
&& (this.major > ver.major
    || this.major == ver.major
        && (this.minor > ver.minor
            || this.minor == ver.minor
                && this.release >= ver.release))
```

Comment (boazavital) [reply affinity/src/main/java/net/openhft/affinity/impl/VersionHelper.java]: figured it could shorter. can update to this if you think it's still readable

**Comment by boazavital** [comment]
Closes #33

**Pull request merged**